src/nlp/load.py

```diff
@@ -530,7 +535,7 @@ def load_dataset(
     """
     ignore_verifications = ignore_verifications or save_infos
     # Download/copy dataset processing script
-    module_path, hash = prepare_module(path, download_config=download_config, dataset=True)
+    module_path, hash = prepare_module(path, download_config=download_config, version=version, dataset=True)
```

Comment:
I guess we could call `prepare_module` with the github url that is already well formatted, instead of passing `version` to `prepare_module` then `cached_path` then `get_from_cache` ? Not sure that `cached_path` should have the `version` logic